src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -121,10 +123,26 @@ private Command parseClientCommands(String input) throws ParseException {
         }
 
         String commandWord = secondaryMatcher.group("commandWord");
-        final String arguments = secondaryMatcher.group("arguments");
+        String arguments = secondaryMatcher.group("arguments");
 
         commandWord = CLIENT_TYPE + " " + commandWord;
+
+        StringTokenizer stringTokenizer = new StringTokenizer(arguments);
```

Comment:
Might be a next TODO which is to abstract away this parsing logic.

Reply:
agree with rayson

Reply:
this was left as is because in the future, we probably should improve the regex pattern matching (as per the screenshot I sent about kang liang's suggestion) 

have added a todo to this to keep note of it